rns_shell/compositor/layers/Layer.cpp

```diff
@@ -173,6 +169,7 @@ void Layer::paint(PaintContext& context) {
 
     SkAutoCanvasRestore save(context.canvas, true); // Save current clip and matrix state.
 
+    context.canvas->setMatrix(baseTransformMatrix_); //Apply transformation matrix before paint
     paintSelf(context); // First paint self and then children if any
     context.canvas->translate(frame_.x(), frame_.y());
```

Comment:
since baseTransformMatrix already considers the parent x,y ; is this canvas translate actually required to be checked.

Reply:
also in maskBounds_ we may have to use absFrame values for clipRect

Reply:
Pointed it rightly, explicit translate not needed, all the transformation is now part of baseTransformMatrix_. Will remove it